beacon-chain/sync/subscriber.go

```diff
@@ -286,6 +286,11 @@ func (s *Service) wrapAndReportValidation(topic string, v wrappedVal) (string, p
 			return pubsub.ValidationIgnore
 		}
 		b, err := v(ctx, pid, msg)
+		// We do not penalize peers if we are hitting pubsub timeouts
+		// trying to process those messages.
+		if b == pubsub.ValidationReject && ctx.Err() != nil {
```

Comment:
Besides timeouts, this also implies context cancelled which is still good. We don't want to Reject and penalize a peer when a context is cancelled